src/ImageSharp/Formats/Tiff/TiffDecoderCore.cs

```diff
@@ -413,7 +423,14 @@ private void DecodeStripsChunky<TPixel>(ImageFrame<TPixel> frame, int rowsPerStr
                     break;
                 }
 
-                decompressor.Decompress(this.inputStream, (uint)stripOffsets[stripIndex], (uint)stripByteCounts[stripIndex], stripHeight, stripBufferSpan);
+                ulong offset = stripOffsets.GetValue(stripIndex) switch { ulong val => val, Number val => (uint)val };
```

Comment:
This conversion will be performed once for each strip, performance will not be affected. Not beautiful, but so far this is the best solution for converting `Number` and `ulong` arrays to same variable (without allocating additional memory, these arrays can be large).

Reply:
@IldarKhayrutdinov Is this comment still relevant? The UI is marked as outdated.

I have no issue extending the type if the extensions can be kept internal?

Reply:
Yes relevant, there was just text formatting. 
I seem to be incorrectly expressed. I meant about changing struct Number, which is a public:
 https://github.com/IldarKhayrutdinov/ImageSharp/blob/978757fb51a3cdc8bbd0a2d8d3df25988ff0e7c4/src/ImageSharp/Primitives/Number.cs#L16-L23

instead
```
private readonly int signedValue;
private readonly uint unsignedValue;
```

required
```
private readonly long signedValue;
private readonly ulong unsignedValue;
```

and add cast operators

Reply:
Since our casting is explicit I would think the expansion is ok.

Reply:
64-bit tags will rarely be used, in Exif it's not used at all, I think it would be better not to change the Number type